File: rust/easy/0001_two_sum.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::new();
        for (i, &el) in nums.iter().enumerate() {
            let needed = target - el;
            if let Some(&idx) = map.get(&needed) {
                return vec![i as i32, idx as i32];
            } else {
                map.insert(el, i);
            }
        }
        unreachable!()
    }
}

File: go/easy/0001_two_sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i, val := range nums {
        if j, ok := m[val]; ok {
            return []int{j, i}
        } else {
            m[target - val] = i
        }
    }
    return []int{}
}
